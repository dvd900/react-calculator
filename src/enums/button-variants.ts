export enum ButtonVariant {
    NUMBER = 'number',
    OPERATION = 'operation',
}
