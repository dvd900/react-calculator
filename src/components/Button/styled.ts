import styled, { css } from 'styled-components'
import { ButtonVariant } from '../../enums/button-variants'
import { COLOR2, COLOR1, COLOR6 } from '../../styles/colors'
import { lgText, smText } from '../../styles/text'
interface StyledButtonProps {
    isSelected: boolean
    variant: ButtonVariant
}

export const StyledButton = styled.button<StyledButtonProps>`
    border: 2px solid black;
    box-shadow: 3px 6px #6a7573;
    :active {
        box-shadow: 2px 3px #535c5a;
        transform: translate(2px, 4px);
    }
    :disabled {
        box-shadow: none;
        filter: brightness(90%);
    }
    :hover:enabled {
        ${(props) =>
            !props.isSelected &&
            css`
                filter: brightness(125%);
            `}
    }

    ${(props) =>
        props.variant === ButtonVariant.NUMBER &&
        css`
            width: 7rem;
            height: 7rem;
            border-radius: 50%;
            ${smText}
            background-color: ${COLOR1};
        `}
    ${(props) =>
        props.isSelected &&
        css`
            background-color: ${COLOR2};
            border: 2px double black;
        `}
    ${(props) =>
        props.variant === ButtonVariant.OPERATION &&
        css`
            ${lgText}
            width: 5rem;
            height: 5rem;
            background-color: ${COLOR6};
        `}
`
