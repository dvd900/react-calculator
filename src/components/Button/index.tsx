import React from 'react'
import { ButtonVariant } from '../../enums/button-variants'
import { StyledButton } from './styled'

interface Props {
    text: string
    variant?: ButtonVariant
    isSelected?: boolean
    isDisabled?: boolean
    onClick?: () => void
}

/* concat styles based on props */
const Button = ({
    text,
    variant = ButtonVariant.NUMBER,
    isSelected = false,
    isDisabled = false,
    onClick,
}: Props) => {
    return (
        <StyledButton
            isSelected={isSelected}
            variant={variant}
            onClick={onClick}
            disabled={isDisabled}
        >
            {text}
        </StyledButton>
    )
}
export default Button
