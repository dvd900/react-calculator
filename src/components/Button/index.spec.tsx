import renderer from 'react-test-renderer'
import Button from '.'
import React from 'react'
import { ButtonVariant } from '../../enums/button-variants'
describe('Button Component', () => {
    const numberButtonProps = {
        text: 'Click Me',
        variant: ButtonVariant.NUMBER,
    }
    const operationButtonProps = {
        text: 'Click Me',
        variant: ButtonVariant.OPERATION,
    }

    it('selected number button variant matches snapshot', () => {
        const tree = renderer.create(<Button {...numberButtonProps} isSelected />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('unselected number button variant matches snapshot', () => {
        const tree = renderer.create(<Button {...numberButtonProps} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('operation button variant matches snapshot', () => {
        const tree = renderer.create(<Button {...operationButtonProps} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('disabled operation button variant matches snapshot', () => {
        const tree = renderer.create(<Button {...operationButtonProps} isDisabled />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
