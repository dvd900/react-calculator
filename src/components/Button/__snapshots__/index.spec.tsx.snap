// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component disabled operation button variant matches snapshot 1`] = `
<button
  className="sc-bdnxRM ljUCIL"
  disabled={true}
>
  Click Me
</button>
`;

exports[`Button Component operation button variant matches snapshot 1`] = `
<button
  className="sc-bdnxRM ljUCIL"
  disabled={false}
>
  Click Me
</button>
`;

exports[`Button Component selected number button variant matches snapshot 1`] = `
<button
  className="sc-bdnxRM gITYun"
  disabled={false}
>
  Click Me
</button>
`;

exports[`Button Component unselected number button variant matches snapshot 1`] = `
<button
  className="sc-bdnxRM kziEhz"
  disabled={false}
>
  Click Me
</button>
`;
